glossary.md

```diff
@@ -147,10 +150,13 @@ Nếu bạn cho rằng một từ không nên dịch ra tiếng Việt, bạn c
 ## I
 | English                    | Tiếng Việt         | Thảo luận tại |
 |----------------------------|--------------------|---------------|
+| implement | xây dựng / lập trình | |
+| implementation | cách xây dựng / lập trình | |
 | implicit feedback          | phản hồi gián tiếp |               |
 | imputation (Preprocessing) | quy buộc           |               |
 | independence assumption    | giả định độc lập   |               |
 | iteration                  | vòng lặp           |               |
+| iterator | bộ duyệt | |
```

Comment:
Mình nghĩ từ này không nên dịch sẽ tốt cho người đọc hơn

Reply:
Mình lại không thấy lí do nào để không dịch cả.

Reply:
@suhohainguyen Để tránh thảo luận lại từ đầu, từ này có thảo luận trong Slack, anh không biết em đã xem qua chưa nhỉ?

Reply:
dạ em thấy trong slack thì tin nhắn gần nhất đề xuất giữ nguyên ấy ạ

Reply:
Chữ iterable, iterator theo mình biết chưa có từ tiếng việt tương đương. 
Hiện tại theo mình biết nó thì nó có 2 cách hiểu, đều ám chỉ việc lặp đi lặp lai một hành động, 
  * thứ nhất trong toán tin, là lặp qua các phần tử trong một tập hợp, (set, array, dict, list)
  * thứ hai trong ngữ cảnh thông dụng là lặp lại qua kết quả của lần chạy trước đó (như kiểu recursion). Cách hiểu này hay thấy trong cách dùng `iteration design`, `iterate quickly through the development cycle`, ...

Nếu cần thiết mình có thể chú thích cách dịch `bộ duyệt` trong ngữ cảnh lập trình thôi.

Reply:
Việc dịch `iterator` là `bộ duyệt` thì mình thấy hoàn toàn hợp lý.

Mình chỉ đang nghĩ nếu sau này có từ `iterable` (khó dịch hơn) mà chẳng may phải để nguyên tiếng anh, thì lúc đó sẽ có 2 khái niệm liên quan chặt chẽ với nhau, nhưng một từ thì được việt hóa, từ còn lại thì không.

Reply:
`iterator` là danh từ. Dịch là `bộ duyệt` cũng khá sát.
`iterable` là tính từ chỉ collection có thể duyệt được, có thể tạm dịch là `khả duyệt` (nghe bệnh bệnh).

Bạn nào có thể trích một số ví dụ có chữ `iterable` không? Thường thì các tính từ kiểu này dùng cho các interface ở mức trừu tượng hơn một cấp so với danh từ (`iterator`). Nên nếu dịch, mình nên tìm cách sao cho diễn đạt được ý là nó trừu tượng hơn một cấp.

Reply:
@tanduong mình vừa ngó qua trong sách này thì có vẻ không có `iterable` rồi, nên chắc lý luận của mình cũng không thuyết phục lắm :smile:
Cơ mà `iterable` lại là [danh từ](https://docs.python.org/3/glossary.html#term-iterable) nhé, thế nên mình mới thấy dịch khó.